### ISSUE TITLE
linter(unicorn/no-for-loop): rule is triggered even if counter is used to index another array

### ISSUE THREAD

**Issue by nekogirls1337** [oxc-project/oxc#11311]
### What version of Oxlint are you using?

0.16.12

### What command did you run?

oxlint

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["unicorn"],
  "rules": {
    "unicorn/no-for-loop": "error"
  }
}
```

### What happened?
```

  × eslint-plugin-unicorn(no-for-loop): Use a `for-of` loop instead of this classical `for` loop.
    ╭─[1.ts:10:9]
 46 │
 47 │         for (let i = 0; i < items.length; i++) {
    ·         ───
 48 │             if (!otherArray[i]) continue
    ╰────
  help: Rewrite the loop as `for (const el of array) { … }`
```

**Comment by camc314** [comment]
hi, thanks for reporting this. Do you mind sharing the full code snippet to reproduce the issue? WIth just those first two lines of the diagnostic, i cannot reproduce this.

**Comment by nekogirls1337** [comment]
@camc314 

```ts
const set = new Set([2])
const items = [1, 2, 3]
const otherArray = []

for (let i = 0; i < items.length; i++) {
    if (!otherArray[i]) continue
    set.add(items[i])
}
```
